htdocs/PI/write/PIWriteRequest.php

```diff
@@ -779,14 +530,482 @@ private function portalIsReadOnly() {
+    if (empty($extensionPropKVArray)) {
+      $extensionPropArray = $service->getServiceProperties()->toArray();
+    } else {
+      $extensionPropArray = array();
+      foreach ($extensionPropKVArray as $key => $value) {
+        $extensionProp = $this->serviceService->getServicePropertyByKeyAndParent($key, $service);
+        if (is_null($extensionProp)) {
+          $this->exceptionWithResponseCode(404,
+            "A property with key \"$key\" could not be found for "
+            . $service->getHostName() . ". No properties have been deleted. "
+          );
+        }
+
+        #If a value has been provided for the property, it needs to matched
+        if (!empty($value) && ($extensionProp->getKeyValue() != $value)) {
+          $this->exceptionWithResponseCode(409,
+            "The value provided for the property with key \"$key\" does " .
+            "not match the existing one. No Properties have been deleted."
+          );
+        }
+
+        $extensionPropArray[] = $extensionProp;
+      }
+    }
+
+    $this->serviceService->deleteServicePropertiesAPI($service, $extensionPropArray, $this->userIdentifierType, $this->userIdentifier);
+  }
+
+  /**
+   * Update an Endpoint
+   *
+   * @param  Site   $siteService
+   * @throws \Exception
+   */
+  private function updateEndpoint (Site $siteService) {
+    #This case requires the serviceService
+    $this->checkServiceServiceSet();
+
+    #Identify the SE
+    try {
+        $endpoint = $this->serviceService->getEndpoint($this->entityID);
+    } catch (\Exception $e) {
+      $this->exceptionWithResponseCode(404, "An endpoint with the specified id could not be found");
+    }
+
+    #Authorisation
+    try {
+        $siteService->checkAuthroisedAPIIDentifier($endpoint->getService()->getParentSite(), $this->userIdentifier, $this->userIdentifierType);
+    } catch(\Exception $e){
+        $this->exceptionWithResponseCode(403, $e->getMessage());
+    }
+
+    #Make the requested change
+    switch($this->entityProperty) {
+        case 'extensionproperties':{
+          $this->updateEndpointExtensionProperties($endpoint);
+          break;
+        }
+        default: {
+          $this->exceptionWithResponseCode(501,$this->genericExceptionMessages["entityTypePropertyCombo"]);
+        }
+    }
+  }
+
+  /**
+   * Update the Extension properties of an endpoint
+   * @param   $endpoint The endpoint to be updated
+   * @throws \Exception
+   */
+  private function updateEndpointExtensionProperties ($endpoint) {
+    #TWO CASES: one there is a single value, the other mutliple - create an array for the single, use array for multiple.
+    if (is_null($this->entityPropertyKey)) {
+        $extensionPropKVArray = $this->entityPropertyKVArray;
+    } else {
+        $extensionPropKVArray = array($this->entityPropertyKey => $this->entityPropertyValue);
+    }
+
+    #Based on Request types run either an add with or without overwritebv or a remove.
```

Comment:
"overwritebv", again?

Reply:
see above

Reply:
overwriteb**y**?

Reply:
overwrite. There is probably a comma missing as well and the entire thing could be phrased better. 

However, you'll notice this was a refactor and the block of code was moved unchanged from elsewhere -this is not an error introduced by this commit, rather it is an error this commit has failed to fix. In that context can we merge this, rather than spell checking every line of code that is touched by it?

Reply:
Touching it means you're now responsible, doesn't it? :stuck_out_tongue_closed_eyes: 

But sure, we can leave it, though I did do a few more suggested changes that could easily be batched up and committed.

Reply:
![image](https://user-images.githubusercontent.com/4125374/67759322-26a4f180-fa37-11e9-9ca1-5da95fb34782.png)

Reply:
That's not all of them. Also, bear in mind that suggested changes can be batched up into a single commit: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes